ucp/_libs/ucx_api.pyx

```diff
@@ -326,6 +332,169 @@ cdef class UCXContext(UCXObject):
         ucp_context_print_info(self._handle, text_fd)
         return decode_text_fd(text_fd)
 
+    def map(self, mem):
+        return UCXMemoryHandle.map(self, mem)
+
+    def alloc(self, size):
+        return UCXMemoryHandle.alloc(self, size)
+
+
+cdef class PackedRemoteKey:
+    cdef void *_key
+    cdef Py_ssize_t _length
+
+    def __cinit__(self, uintptr_t packed_key_as_int, Py_ssize_t length):
+        key = <void *> packed_key_as_int
+        self._key = malloc(length)
+        self._length = length
+        memcpy(self._key, key, length)
+
+    @classmethod
+    def from_buffer(cls, buffer):
+        buf = Array(buffer)
+        assert buf.c_contiguous
+        return PackedRemoteKey(buf.ptr, buf.nbytes)
+
+    @classmethod
+    def from_memh(self, UCXMemoryHandle mem):
+        cdef void *key
+        cdef size_t len
+        cdef ucs_status_t status
+        status = ucp_rkey_pack(mem._context._handle, mem._memh, &key, &len)
+        assert_ucs_status(status)
+        return PackedRemoteKey(<uintptr_t>key, len)
+
+    def __dealloc__(self):
+        ucp_rkey_buffer_release(self._key)
+
+    @property
+    def key(self):
+        return int(<uintptr_t><void*>self._key)
+
+    @property
+    def length(self):
+        return int(self._length)
+
+    def __getbuffer__(self, Py_buffer *buffer, int flags):
+        if (flags & PyBUF_WRITABLE) == PyBUF_WRITABLE:
+            raise BufferError("Requested writable view on readonly data")
+        buffer.buf = <void*>self._key
+        buffer.obj = self
+        buffer.len = self._length
+        buffer.readonly = True
+        buffer.itemsize = 1
+        if (flags & PyBUF_FORMAT) == PyBUF_FORMAT:
+            buffer.format = b"B"
+        else:
+            buffer.format = NULL
+        buffer.ndim = 1
+        if (flags & PyBUF_ND) == PyBUF_ND:
+            buffer.shape = &self._length
+        else:
+            buffer.shape = NULL
+        buffer.strides = NULL
+        buffer.suboffsets = NULL
+        buffer.internal = NULL
+
+    def __releasebuffer__(self, Py_buffer *buffer):
+        pass
+
+    def __reduce__(self):
+        return (PackedRemoteKey.from_buffer, (bytes(self),))
+
+    def __hash__(self):
+        return hash(bytes(self))
+
+
+def _ucx_mem_handle_finalizer(uintptr_t handle_as_int, UCXContext ctx):
+    assert ctx.initialized
+    cdef ucp_mem_h handle = <ucp_mem_h>handle_as_int
+    cdef ucs_status_t status
+    status = ucp_mem_unmap(ctx._handle, handle)
+    assert_ucs_status(status)
+
+
+cdef class UCXMemoryHandle(UCXObject):
```

Comment:
I would be good with a description of what this class represent and its parameters.